### DISCUSSION TITLE
React on code-server

### DISCUSSION THREAD

**Discussion by iMacX46** [coder/code-server#4229] [Q&A] [answered]
Hi!
I have a problem with code-server deployed on railway.
Why can't I launch the command npx create-react-app on terminal?

**Comment by jsjoeio** [comment, marked as answer]
What happens when you run that? Do you have Node/npm installed in your deployment?

Also [read this](https://coder.com/docs/code-server/v3.12.0/guide#proxying-to-create-a-react-app) for running React apps.

**Reply by iMacX46** [reply]
Yes I have:
-Node v14.17.6
-NPM v6.14.15

And I obtain the error on the image below:

![image](https://user-images.githubusercontent.com/52834736/134553524-92c82b7c-3efe-459b-963e-7a9ada9c4563.png)

**Reply by jsjoeio** [reply]
This seems to be an issue with either Node/npm or Railway. Here are two solutions I found:
- [option 1](https://stackoverflow.com/a/60192748/3015595)
- [option 2](https://github.com/facebook/create-react-app/issues/8896#issuecomment-619406384)

Hope that helps!

**Reply by iMacX46** [reply]
I solved using yarn instead of npm

- sudo npm install -g yarn
- yarn global add create-react-app
- sudo npx create-react-app my-app

Thanks a lot!  👍

**Reply by jsjoeio** [reply]
Hooray! No problem!
